### ISSUE TITLE
`bevy-glsl-to-spirv` is 38 MB of source code

### ISSUE THREAD

**Issue by jyn514** [bevyengine/bevy#1584]
## Bevy version

f61e44db2824c0cc4b4815748c4d3f051002ada7

## Operating system & version

Ubuntu 20.04

## What you did

`git clone https://github.com/bevyengine/bevy && cargo check`

## What you expected to happen

Downloading bevy sources takes very little time.

## What actually happened

Cargo takes over 38 MB to download the source code alone.
```
Downloaded 62 crates (48.4 MB) in 9.61s (largest was `bevy-glsl-to-spirv` at 37.9 MB)
```

**Comment by jyn514** [comment]
I haven't looked into why it's so big, but 38 MB seems pretty giant.

**Comment by GrygrFlzr** [comment]
The problem is that it’s delivering literal binaries. It’s possible to always compile from source, but that brings a penalty to compilation times.

**Comment by mockersf** [comment]
You can find more information, and the path forward for that in #1073 

TL;DR:
- short term: provide easer support to [`shaderc`](https://crates.io/crates/shaderc)
- long term: replace bevy-glsl-to-spirv by [`naga`](https://crates.io/crates/naga)
- hypothetical longer term? support [`rust-gpu`](https://github.com/EmbarkStudios/rust-gpu)

**Comment by james7132** [comment]
Closing this as all shader transpilation is now done via naga, as of the 0.6 render rework.
